libcipher/caesar.py

```diff
@@ -0,0 +1,108 @@
+__author__ = 'kskrishnasangeeth'
+
+
+class Error(Exception):
+    pass
+
+
+class WrongKey(Error):
+    """ class created for throwing Exception in the event of wrong key being supplied
+
+    Attributes:
+        msg (str): Explanation on why key is not allowed.
+        key (int): Key that is supplied.
+
+    Args:
+        msg (str): Explanation on why key is not allowed.
+        key (int): Key that is supplied.
+
+   """
+    def __init__(self, msg, key):
+        self.msg = msg
+        self.key = key
+
+
+class NoKeyGiven(Error):
+    """ class created for throwing Exception in the event of no key being supplied
+
+    Attributes:
+        msg (str): Message to provide key for encryption.
+
+    Args:
+        msg (str): Message to provide key for encryption.
+
+    """
+    def __init__(self, msg):
+        self.msg = msg
+
+
+def cipher_helper(message, key):
+    """Function for modularising and removing some of the repeated code.
+
+    Args:
+        message (str): Message that is supplied for encryption/decryption.
+        key (int): Key used for encryption/decryption.
+
+    Returns:
+        message (str): Message in lowercase.
+
+    """
+    if key is None:
+        raise NoKeyGiven("Key is needed for Encryption.")
+
+    if key <= 0 or key >= 27:
+        raise WrongKey("Key Value should be between 0 and 26. Got Key as", key)
+
+    return message.lower()
+
+
+def encrypt(message, key=None):
```

Comment:
@bravegnu Few more thoughts on this.

The reason i made <code>key</code> initialized to *None* at first place is when we do this 

<code>>>>encrypt("message") </code>
I get this as the error message.

```Python
TypeError: encrypt() missing 1 required positional argument: 'key'
```
which is the default error message.

So my custom Exception message becomes pointless as we really don't do things like
<code>encrypt("hello",None)</code>  in real scenarios  to get the custom message below

```Python
    raise NoKeyGiven("Key is needed to be entered")
```
So it leaves me with two options 
a) Remove the exception for missing Key
b) Leave the key as optional in <code>encrypt()</code> and <code>decrypt()</code> functions. 

What do you reckon is the best thing to do?

> Point 2, do you really have to check for key is None. And why stop there and why not check message is None.

This will happen only when we try something like this right

<code>>>>encrypt( ,10) </code>

This appears to be syntactically wrong and i guess we don't need to have an exception for it.

Or a scenario where message is missing could be <code>>>>encrypt(10)</code> , but i am not sure if this scenario is worth pursuing.